### ISSUE TITLE
Kuberenetes downstream module dependencies require updating protobuf to v1.4.x, which prints protoc-gen-go/generator deprecation warnings

### ISSUE THREAD

**Issue by brandond** [k3s-io/k3s#3195]
* Kubernetes v1.21.0 bumped cadvisor to v0.39.0: https://github.com/kubernetes/kubernetes/blob/v1.21.0/go.mod#L52
    * cadvisor CANNOT be held back, as kubelet needs some stuff that's only in v0.39.0
* cadvisor v0.39.0 needs github.com/moby/sys/mountinfo v0.4.0: https://github.com/google/cadvisor/commit/b0134056e91f50738bc258e631a74e26f64c9a86
* runc needs github.com/moby/sys/mountinfo v0.1.3 up until v1.0.0-rc93: https://github.com/opencontainers/runc/commit/17de6f803d161f93f851e62296fc51f6d484bd50
* Runc v1.0.0-rc93 needs github.com/checkpoint-restore/go-criu v5 : https://github.com/opencontainers/runc/commit/97f2e351a89587e0c58a10269b384216a959fc8d
* go-criu v5 needs github.com/golang/protobuf v1.4.x: https://github.com/checkpoint-restore/go-criu/blob/v5.0.0/rpc/rpc.pb.go#L26
* protobuf v1.4.x prints a warning whenever any binary is run about protoc-gen-go/generator being deprecated: https://github.com/golang/protobuf/commit/d3786a266248bfc63bd81d943ed4af6fb2f6b5c2
    * Everything SEEMS to work fine with protobuf v1.4.x, except for the warning that it spams every time you run kubectl or any other k3s-wrapped command.

What is using protoc-gen-go/generator to trigger the warning?
* grpc-gateway uses protoc-gen-go/generator up until v2: https://github.com/grpc-ecosystem/grpc-gateway/blob/v1/runtime/fieldmask.go#L9
* grpc-gateway v1 is still used by etcd v3.4 and won't go away until v3.5: https://github.com/etcd-io/etcd/blob/v3.4.15/embed/serve.go#L41

tl;dr we need to fork protobuf v1.4.x to silence the [protoc-gen-go/generator warning](https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L44-L46), since we're stuck with it due to etcd. The warning is triggered whenever the package is linked in, not just when you actually use it.

**Comment by brandond** [comment]
Closing, nothing for QA to test on this one.
